ICE : `Region parameter out of range` when instantiating in region

<!--
Thank you for finding an Internal Compiler Error! 🧊  If possible, try to provide
a minimal verifiable example. You can read "Rust Bug Minimization Patterns" for
how to create smaller examples.
http://blog.pnkfx.org/blog/2019/11/18/rust-bug-minimization-patterns/
-->

### Code

(reduced)
```Rust
trait Trait<T: ?Sized> {
    const C<'a>: &'a T
}

struct Implementor;

impl Trait<str> for Implementor {
    const C<>: &'a str = "C"
}

```


(original)
```Rust
#![feature(generic_const_items)]
#![allow(incomplete_features)]

pub const K<'a, T: 'a + Copy, const N: usize>: Option<[T; N]> = None
where
    String: From<T>;

pub trait Trait<T: ?Sized> {
    const C<'a>: &'a T
    where
        T: 'a + Eq;
}

pub struct Implementor;

impl Trait<str> for Implementor {
    const C<>: &'a str = "C"
    // In real code we could've left off this bound but adding it explicitly allows us to test if
    // we render where-clauses on associated consts inside impl blocks correctly.
    where
        str: 'a;
}

```


### Meta
<!--
If you're using the stable version of the compiler, you should also check if the
bug also exists in the beta or nightly versions.
-->

`rustc --version --verbose`:
```
rustc 1.79.0-nightly (9d5cdf75a 2024-04-07)
binary: rustc
commit-hash: 9d5cdf75aa42faaf0b58ba21a510117e8d0051a3
commit-date: 2024-04-07
host: x86_64-apple-darwin
release: 1.79.0-nightly
LLVM version: 18.1.3
```

### Error output

```
error: expected `;`, found `}`
 --> ./743F0725FEABD9E7F4F5425F8913337F0F0429BC495C842AFD6DCFB4EE1922E0.rs:2:23
  |
2 |     const C<'a>: &'a T
  |                       ^ help: add `;` here
3 | }
  | - unexpected token

error: expected `;`, found `}`
 --> ./743F0725FEABD9E7F4F5425F8913337F0F0429BC495C842AFD6DCFB4EE1922E0.rs:8:29
  |
8 |     const C<>: &'a str = "C"
  |                             ^ help: add `;` here
9 | }
  | - unexpected token

error[E0261]: use of undeclared lifetime name `'a`
 --> ./743F0725FEABD9E7F4F5425F8913337F0F0429BC495C842AFD6DCFB4EE1922E0.rs:8:17
  |
7 | impl Trait<str> for Implementor {
  |     - help: consider introducing lifetime `'a` here: `<'a>`
8 |     const C<>: &'a str = "C"
  |                 ^^ undeclared lifetime

error[E0658]: generic const items are experimental
 --> ./743F0725FEABD9E7F4F5425F8913337F0F0429BC495C842AFD6DCFB4EE1922E0.rs:2:12
  |
2 |     const C<'a>: &'a T
  |            ^^^^
  |
  = note: see issue #113521 <https://github.com/rust-lang/rust/issues/113521> for more information
  = help: add `#![feature(generic_const_items)]` to the crate attributes to enable
  = note: this compiler was built on 2024-04-11; consider upgrading it if it is out of date

error[E0658]: generic const items are experimental
 --> ./743F0725FEABD9E7F4F5425F8913337F0F0429BC495C842AFD6DCFB4EE1922E0.rs:8:12
  |
8 |     const C<>: &'a str = "C"
  |            ^^
  |
  = note: see issue #113521 <https://github.com/rust-lang/rust/issues/113521> for more information
  = help: add `#![feature(generic_const_items)]` to the crate attributes to enable
  = note: this compiler was built on 2024-04-11; consider upgrading it if it is out of date

error[E0601]: `main` function not found in crate `743F0725FEABD9E7F4F5425F8913337F0F0429BC495C842AFD6DCFB4EE1922E0`
 --> ./743F0725FEABD9E7F4F5425F8913337F0F0429BC495C842AFD6DCFB4EE1922E0.rs:9:2
  |
9 | }
  |  ^ consider adding a `main` function to `./743F0725FEABD9E7F4F5425F8913337F0F0429BC495C842AFD6DCFB4EE1922E0.rs`
```

<!--
Include a backtrace in the code block by setting `RUST_BACKTRACE=1` in your
environment. E.g. `RUST_BACKTRACE=1 cargo build`.
-->
<details><summary><strong>Backtrace</strong></summary>
<p>

```
error: internal compiler error: compiler/rustc_middle/src/ty/generic_args.rs:825:13: Region parameter out of range when instantiating in region 'a (index=2, args = [Implementor, str])

thread 'rustc' panicked at compiler/rustc_middle/src/ty/generic_args.rs:825:13:
Box<dyn Any>
stack backtrace:
   0:        0x1030abe13 - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::hcb82b7434a7ad069
   1:        0x1030f7a8b - core::fmt::write::h20ce779d62646bbc
   2:        0x1030a1f5e - std::io::Write::write_fmt::hc9677006e9d09a01
   3:        0x1030abc01 - std::sys_common::backtrace::print::h574caf35b482d642
   4:        0x1030aea69 - std::panicking::default_hook::{{closure}}::hd4be3fc55f74a05f
   5:        0x1030ae7b0 - std::panicking::default_hook::h2f05dfe1d561d141
   6:        0x10bd0728c - std[49a09ef43dd137fc]::panicking::update_hook::<alloc[5c0aec220d3fb6dd]::boxed::Box<rustc_driver_impl[9cc6f343bed42afe]::install_ice_hook::{closure#0}>>::{closure#0}
   7:        0x1030af58c - std::panicking::rust_panic_with_hook::hf0bdb32e38a98365
   8:        0x10bd72e8a - std[49a09ef43dd137fc]::panicking::begin_panic::<rustc_errors[eb8a5b66a8ece974]::ExplicitBug>::{closure#0}
   9:        0x10bd6b3b9 - std[49a09ef43dd137fc]::sys_common::backtrace::__rust_end_short_backtrace::<std[49a09ef43dd137fc]::panicking::begin_panic<rustc_errors[eb8a5b66a8ece974]::ExplicitBug>::{closure#0}, !>
  10:        0x11049f6a9 - std[49a09ef43dd137fc]::panicking::begin_panic::<rustc_errors[eb8a5b66a8ece974]::ExplicitBug>
  11:        0x10bd865d6 - <rustc_errors[eb8a5b66a8ece974]::diagnostic::BugAbort as rustc_errors[eb8a5b66a8ece974]::diagnostic::EmissionGuarantee>::emit_producing_guarantee
  12:        0x10c86250e - rustc_middle[86d1f27645b76717]::util::bug::opt_span_bug_fmt::<rustc_span[c866a394ab29ee91]::span_encoding::Span>::{closure#0}
  13:        0x10c812a67 - rustc_middle[86d1f27645b76717]::ty::context::tls::with_opt::<rustc_middle[86d1f27645b76717]::util::bug::opt_span_bug_fmt<rustc_span[c866a394ab29ee91]::span_encoding::Span>::{closure#0}, !>::{closure#0}
  14:        0x10c8124d5 - rustc_middle[86d1f27645b76717]::ty::context::tls::with_context_opt::<rustc_middle[86d1f27645b76717]::ty::context::tls::with_opt<rustc_middle[86d1f27645b76717]::util::bug::opt_span_bug_fmt<rustc_span[c866a394ab29ee91]::span_encoding::Span>::{closure#0}, !>::{closure#0}, !>
  15:        0x11055e13b - rustc_middle[86d1f27645b76717]::util::bug::bug_fmt
  16:        0x11055dca4 - <rustc_middle[86d1f27645b76717]::ty::generic_args::ArgFolder as rustc_type_ir[8aa2137bd864a930]::fold::TypeFolder<rustc_middle[86d1f27645b76717]::ty::context::TyCtxt>>::fold_region::region_param_out_of_range
  17:        0x10c840ff2 - <rustc_middle[86d1f27645b76717]::ty::generic_args::ArgFolder as rustc_type_ir[8aa2137bd864a930]::fold::TypeFolder<rustc_middle[86d1f27645b76717]::ty::context::TyCtxt>>::fold_region
  18:        0x10c84cec1 - <rustc_middle[86d1f27645b76717]::ty::Ty as rustc_type_ir[8aa2137bd864a930]::fold::TypeSuperFoldable<rustc_middle[86d1f27645b76717]::ty::context::TyCtxt>>::try_super_fold_with::<rustc_middle[86d1f27645b76717]::ty::generic_args::ArgFolder>
  19:        0x10bfb59e2 - rustc_hir_analysis[90609d44e62386d1]::check::compare_impl_item::compare_impl_const_raw
  20:        0x10d15c2c2 - rustc_query_impl[16231b3ed4d194e2]::plumbing::__rust_begin_short_backtrace::<rustc_query_impl[16231b3ed4d194e2]::query_impl::compare_impl_const::dynamic_query::{closure#2}::{closure#0}, rustc_middle[86d1f27645b76717]::query::erase::Erased<[u8; 1usize]>>
  21:        0x10d1089c5 - <rustc_query_impl[16231b3ed4d194e2]::query_impl::compare_impl_const::dynamic_query::{closure#2} as core[75211c3caa4ba820]::ops::function::FnOnce<(rustc_middle[86d1f27645b76717]::ty::context::TyCtxt, (rustc_span[c866a394ab29ee91]::def_id::LocalDefId, rustc_span[c866a394ab29ee91]::def_id::DefId))>>::call_once
  22:        0x10cfbf8d7 - rustc_query_system[acfab0ae1d333eea]::query::plumbing::try_execute_query::<rustc_query_impl[16231b3ed4d194e2]::DynamicConfig<rustc_query_system[acfab0ae1d333eea]::query::caches::DefaultCache<(rustc_span[c866a394ab29ee91]::def_id::LocalDefId, rustc_span[c866a394ab29ee91]::def_id::DefId), rustc_middle[86d1f27645b76717]::query::erase::Erased<[u8; 1usize]>>, false, false, false>, rustc_query_impl[16231b3ed4d194e2]::plumbing::QueryCtxt, false>
  23:        0x10d1ce84c - rustc_query_impl[16231b3ed4d194e2]::query_impl::compare_impl_const::get_query_non_incr::__rust_end_short_backtrace
  24:        0x10bf999e3 - rustc_hir_analysis[90609d44e62386d1]::check::check::check_impl_items_against_trait
  25:        0x10bf925ab - rustc_hir_analysis[90609d44e62386d1]::check::check::check_item_type
  26:        0x10bfca9b4 - rustc_hir_analysis[90609d44e62386d1]::check::wfcheck::check_well_formed
  27:        0x10d15bdfa - rustc_query_impl[16231b3ed4d194e2]::plumbing::__rust_begin_short_backtrace::<rustc_query_impl[16231b3ed4d194e2]::query_impl::check_well_formed::dynamic_query::{closure#2}::{closure#0}, rustc_middle[86d1f27645b76717]::query::erase::Erased<[u8; 1usize]>>
  28:        0x10cfdd77c - rustc_query_system[acfab0ae1d333eea]::query::plumbing::try_execute_query::<rustc_query_impl[16231b3ed4d194e2]::DynamicConfig<rustc_query_system[acfab0ae1d333eea]::query::caches::VecCache<rustc_hir[427244ed39142342]::hir_id::OwnerId, rustc_middle[86d1f27645b76717]::query::erase::Erased<[u8; 1usize]>>, false, false, false>, rustc_query_impl[16231b3ed4d194e2]::plumbing::QueryCtxt, false>
  29:        0x10d1a3ca0 - rustc_query_impl[16231b3ed4d194e2]::query_impl::check_well_formed::get_query_non_incr::__rust_end_short_backtrace
  30:        0x10bf1c612 - rustc_middle[86d1f27645b76717]::query::plumbing::query_ensure_error_guaranteed::<rustc_query_system[acfab0ae1d333eea]::query::caches::VecCache<rustc_hir[427244ed39142342]::hir_id::OwnerId, rustc_middle[86d1f27645b76717]::query::erase::Erased<[u8; 1usize]>>, ()>
  31:        0x10bfd8781 - rustc_hir_analysis[90609d44e62386d1]::check::wfcheck::check_mod_type_wf
  32:        0x10d15bdea - rustc_query_impl[16231b3ed4d194e2]::plumbing::__rust_begin_short_backtrace::<rustc_query_impl[16231b3ed4d194e2]::query_impl::check_mod_type_wf::dynamic_query::{closure#2}::{closure#0}, rustc_middle[86d1f27645b76717]::query::erase::Erased<[u8; 1usize]>>
  33:        0x10cf90da1 - rustc_query_system[acfab0ae1d333eea]::query::plumbing::try_execute_query::<rustc_query_impl[16231b3ed4d194e2]::DynamicConfig<rustc_query_system[acfab0ae1d333eea]::query::caches::DefaultCache<rustc_span[c866a394ab29ee91]::def_id::LocalModDefId, rustc_middle[86d1f27645b76717]::query::erase::Erased<[u8; 1usize]>>, false, false, false>, rustc_query_impl[16231b3ed4d194e2]::plumbing::QueryCtxt, false>
  34:        0x10d180d70 - rustc_query_impl[16231b3ed4d194e2]::query_impl::check_mod_type_wf::get_query_non_incr::__rust_end_short_backtrace
  35:        0x10c05269c - rustc_hir_analysis[90609d44e62386d1]::check_crate
  36:        0x10c4c5672 - rustc_interface[6612d49bd5b18005]::passes::run_required_analyses
  37:        0x10c4c7c90 - rustc_interface[6612d49bd5b18005]::passes::analysis
  38:        0x10d15eb6a - rustc_query_impl[16231b3ed4d194e2]::plumbing::__rust_begin_short_backtrace::<rustc_query_impl[16231b3ed4d194e2]::query_impl::analysis::dynamic_query::{closure#2}::{closure#0}, rustc_middle[86d1f27645b76717]::query::erase::Erased<[u8; 1usize]>>
  39:        0x10cf4405e - rustc_query_system[acfab0ae1d333eea]::query::plumbing::try_execute_query::<rustc_query_impl[16231b3ed4d194e2]::DynamicConfig<rustc_query_system[acfab0ae1d333eea]::query::caches::SingleCache<rustc_middle[86d1f27645b76717]::query::erase::Erased<[u8; 1usize]>>, false, false, false>, rustc_query_impl[16231b3ed4d194e2]::plumbing::QueryCtxt, false>
  40:        0x10d168f37 - rustc_query_impl[16231b3ed4d194e2]::query_impl::analysis::get_query_non_incr::__rust_end_short_backtrace
  41:        0x10bcb93b7 - <rustc_interface[6612d49bd5b18005]::queries::QueryResult<&rustc_middle[86d1f27645b76717]::ty::context::GlobalCtxt>>::enter::<core[75211c3caa4ba820]::result::Result<(), rustc_span[c866a394ab29ee91]::ErrorGuaranteed>, rustc_driver_impl[9cc6f343bed42afe]::run_compiler::{closure#0}::{closure#1}::{closure#3}>
  42:        0x10bd0e52f - rustc_interface[6612d49bd5b18005]::interface::run_compiler::<core[75211c3caa4ba820]::result::Result<(), rustc_span[c866a394ab29ee91]::ErrorGuaranteed>, rustc_driver_impl[9cc6f343bed42afe]::run_compiler::{closure#0}>::{closure#0}
  43:        0x10bcfa2cd - std[49a09ef43dd137fc]::sys_common::backtrace::__rust_begin_short_backtrace::<rustc_interface[6612d49bd5b18005]::util::run_in_thread_with_globals<rustc_interface[6612d49bd5b18005]::util::run_in_thread_pool_with_globals<rustc_interface[6612d49bd5b18005]::interface::run_compiler<core[75211c3caa4ba820]::result::Result<(), rustc_span[c866a394ab29ee91]::ErrorGuaranteed>, rustc_driver_impl[9cc6f343bed42afe]::run_compiler::{closure#0}>::{closure#0}, core[75211c3caa4ba820]::result::Result<(), rustc_span[c866a394ab29ee91]::ErrorGuaranteed>>::{closure#0}, core[75211c3caa4ba820]::result::Result<(), rustc_span[c866a394ab29ee91]::ErrorGuaranteed>>::{closure#0}::{closure#0}, core[75211c3caa4ba820]::result::Result<(), rustc_span[c866a394ab29ee91]::ErrorGuaranteed>>
  44:        0x10bd143e6 - <<std[49a09ef43dd137fc]::thread::Builder>::spawn_unchecked_<rustc_interface[6612d49bd5b18005]::util::run_in_thread_with_globals<rustc_interface[6612d49bd5b18005]::util::run_in_thread_pool_with_globals<rustc_interface[6612d49bd5b18005]::interface::run_compiler<core[75211c3caa4ba820]::result::Result<(), rustc_span[c866a394ab29ee91]::ErrorGuaranteed>, rustc_driver_impl[9cc6f343bed42afe]::run_compiler::{closure#0}>::{closure#0}, core[75211c3caa4ba820]::result::Result<(), rustc_span[c866a394ab29ee91]::ErrorGuaranteed>>::{closure#0}, core[75211c3caa4ba820]::result::Result<(), rustc_span[c866a394ab29ee91]::ErrorGuaranteed>>::{closure#0}::{closure#0}, core[75211c3caa4ba820]::result::Result<(), rustc_span[c866a394ab29ee91]::ErrorGuaranteed>>::{closure#1} as core[75211c3caa4ba820]::ops::function::FnOnce<()>>::call_once::{shim:vtable#0}
  45:        0x1030b8ccb - std::sys::pal::unix::thread::Thread::new::thread_start::h6978137b7cf37fba
  46:     0x7ff818270202 - __pthread_start

note: we would appreciate a bug report: https://github.com/rust-lang/rust/issues/new?labels=C-bug%2C+I-ICE%2C+T-compiler&template=ice.md

note: please make sure that you have updated to the latest nightly

note: please attach the file at `/Users/sal/Documents/240410버전(240412검토)/rustc-ice-2024-04-12T01_35_50-84190.txt` to your bug report

query stack during panic:
#0 [compare_impl_const] checking assoc const `<impl at ./743F0725FEABD9E7F4F5425F8913337F0F0429BC495C842AFD6DCFB4EE1922E0.rs:7:1: 7:32>::C` has the same type as trait item
#1 [check_well_formed] checking that `<impl at ./743F0725FEABD9E7F4F5425F8913337F0F0429BC495C842AFD6DCFB4EE1922E0.rs:7:1: 7:32>` is well-formed
#2 [check_mod_type_wf] checking that types are well-formed in top-level module
#3 [analysis] running analysis passes on this crate
end of query stack
error: aborting due to 7 previous errors

```

</p>
</details>

--- a/compiler/rustc_hir_analysis/src/check/compare_impl_item.rs
+++ b/compiler/rustc_hir_analysis/src/check/compare_impl_item.rs
@@ -1723,6 +1723,7 @@ pub(super) fn compare_impl_const_raw(
 
     compare_number_of_generics(tcx, impl_const_item, trait_const_item, false)?;
     compare_generic_param_kinds(tcx, impl_const_item, trait_const_item, false)?;
+    check_region_bounds_on_impl_item(tcx, impl_const_item, trait_const_item, false)?;
     compare_const_predicate_entailment(tcx, impl_const_item, trait_const_item, impl_trait_ref)
 }
 
@@ -1762,8 +1763,6 @@ fn compare_const_predicate_entailment<'tcx>(
 
     let impl_ct_predicates = tcx.predicates_of(impl_ct.def_id);
     let trait_ct_predicates = tcx.predicates_of(trait_ct.def_id);
-
-    check_region_bounds_on_impl_item(tcx, impl_ct, trait_ct, false)?;
 
     // The predicates declared by the impl definition, the trait and the
     // associated const in the trait are assumed.
@@ -1866,6 +1865,7 @@ pub(super) fn compare_impl_ty<'tcx>(
     let _: Result<(), ErrorGuaranteed> = try {
         compare_number_of_generics(tcx, impl_ty, trait_ty, false)?;
         compare_generic_param_kinds(tcx, impl_ty, trait_ty, false)?;
+        check_region_bounds_on_impl_item(tcx, impl_ty, trait_ty, false)?;
         compare_type_predicate_entailment(tcx, impl_ty, trait_ty, impl_trait_ref)?;
         check_type_bounds(tcx, trait_ty, impl_ty, impl_trait_ref)?;
     };
@@ -1885,8 +1885,6 @@ fn compare_type_predicate_entailment<'tcx>(
 
     let impl_ty_predicates = tcx.predicates_of(impl_ty.def_id);
     let trait_ty_predicates = tcx.predicates_of(trait_ty.def_id);
-
-    check_region_bounds_on_impl_item(tcx, impl_ty, trait_ty, false)?;
 
     let impl_ty_own_bounds = impl_ty_predicates.instantiate_own(tcx, impl_args);
     if impl_ty_own_bounds.len() == 0 {

--- a/tests/ui/generic-const-items/compare-impl-item.rs
+++ b/tests/ui/generic-const-items/compare-impl-item.rs
@@ -6,9 +6,10 @@ trait Trait<P> {
     const B<const K: u64, const Q: u64>: u64;
     const C<T>: T;
     const D<const N: usize>: usize;
+    const E<'a>: &'a ();
 
-    const E: usize;
-    const F<T: PartialEq>: ();
+    const F: usize;
+    const G<T: PartialEq>: ();
 }
 
 impl<P> Trait<P> for () {
@@ -20,11 +21,13 @@ impl<P> Trait<P> for () {
     //~^ ERROR const `C` has 0 type parameters but its trait declaration has 1 type parameter
     const D<const N: u16>: u16 = N;
     //~^ ERROR const `D` has an incompatible generic parameter for trait `Trait`
+    const E: &'static () = &();
+    //~^ ERROR lifetime parameters or bounds on const `E` do not match the trait declaration
 
-    const E: usize = 1024
+    const F: usize = 1024
     where
         P: Copy; //~ ERROR impl has stricter requirements than trait
-    const F<T: Eq>: () = (); //~ ERROR impl has stricter requirements than trait
+    const G<T: Eq>: () = (); //~ ERROR impl has stricter requirements than trait
 }
 
 fn main() {}

--- a/tests/ui/generic-const-items/compare-impl-item.stderr
+++ b/tests/ui/generic-const-items/compare-impl-item.stderr
@@ -1,5 +1,5 @@
 error[E0049]: const `A` has 1 type parameter but its trait declaration has 0 type parameters
-  --> $DIR/compare-impl-item.rs:15:13
+  --> $DIR/compare-impl-item.rs:16:13
    |
 LL |     const A: ();
    |            - expected 0 type parameters
@@ -8,7 +8,7 @@ LL |     const A<T>: () = ();
    |             ^ found 1 type parameter
 
 error[E0049]: const `B` has 1 const parameter but its trait declaration has 2 const parameters
-  --> $DIR/compare-impl-item.rs:17:13
+  --> $DIR/compare-impl-item.rs:18:13
    |
 LL |     const B<const K: u64, const Q: u64>: u64;
    |             ------------  ------------
@@ -19,7 +19,7 @@ LL |     const B<const K: u64>: u64 = 0;
    |             ^^^^^^^^^^^^ found 1 const parameter
 
 error[E0049]: const `C` has 0 type parameters but its trait declaration has 1 type parameter
-  --> $DIR/compare-impl-item.rs:19:13
+  --> $DIR/compare-impl-item.rs:20:13
    |
 LL |     const C<T>: T;
    |             - expected 1 type parameter
@@ -28,7 +28,7 @@ LL |     const C<'a>: &'a str = "";
    |             ^^ found 0 type parameters
 
 error[E0053]: const `D` has an incompatible generic parameter for trait `Trait`
-  --> $DIR/compare-impl-item.rs:21:13
+  --> $DIR/compare-impl-item.rs:22:13
    |
 LL | trait Trait<P> {
    |       -----
@@ -42,25 +42,34 @@ LL | impl<P> Trait<P> for () {
 LL |     const D<const N: u16>: u16 = N;
    |             ^^^^^^^^^^^^ found const parameter of type `u16`
 
-error[E0276]: impl has stricter requirements than trait
-  --> $DIR/compare-impl-item.rs:26:12
+error[E0195]: lifetime parameters or bounds on const `E` do not match the trait declaration
+  --> $DIR/compare-impl-item.rs:24:12
    |
-LL |     const E: usize;
-   |     -------------- definition of `E` from trait
+LL |     const E<'a>: &'a ();
+   |            ---- lifetimes in impl do not match this const in trait
+...
+LL |     const E: &'static () = &();
+   |            ^ lifetimes do not match const in trait
+
+error[E0276]: impl has stricter requirements than trait
+  --> $DIR/compare-impl-item.rs:29:12
+   |
+LL |     const F: usize;
+   |     -------------- definition of `F` from trait
 ...
 LL |         P: Copy;
    |            ^^^^ impl has extra requirement `P: Copy`
 
 error[E0276]: impl has stricter requirements than trait
-  --> $DIR/compare-impl-item.rs:27:16
+  --> $DIR/compare-impl-item.rs:30:16
    |
-LL |     const F<T: PartialEq>: ();
-   |     ------------------------- definition of `F` from trait
+LL |     const G<T: PartialEq>: ();
+   |     ------------------------- definition of `G` from trait
 ...
-LL |     const F<T: Eq>: () = ();
+LL |     const G<T: Eq>: () = ();
    |                ^^ impl has extra requirement `T: Eq`
 
-error: aborting due to 6 previous errors
+error: aborting due to 7 previous errors
 
-Some errors have detailed explanations: E0049, E0053, E0276.
+Some errors have detailed explanations: E0049, E0053, E0195, E0276.
 For more information about an error, try `rustc --explain E0049`.

